tests/unit/scrapy/utils/test_apply_apify_settings.py

```diff
@@ -0,0 +1,109 @@
+            }
+        ),
+    ),
+    # Test case with default settings and proxy configuration
+    TestCase(
+        settings=Settings(),
+        proxy_config={'useApifyProxy': True, 'apifyProxyGroups': []},
+        expected_output=Settings(
+            {
+                'SCHEDULER': 'apify.scrapy.scheduler.ApifyScheduler',
+                'ITEM_PIPELINES': {
+                    'apify.scrapy.pipelines.ActorDatasetPushPipeline': 1000,
+                },
+                'DOWNLOADER_MIDDLEWARES': {
+                    'scrapy.downloadermiddlewares.robotstxt.RobotsTxtMiddleware': None,
+                    'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': None,
+                    'scrapy.downloadermiddlewares.retry.RetryMiddleware': None,
+                    'apify.scrapy.middlewares.ApifyHttpProxyMiddleware': 950,
+                    'apify.scrapy.middlewares.ApifyRetryMiddleware': 1000,
+                },
+                'APIFY_PROXY_SETTINGS': {'useApifyProxy': True, 'apifyProxyGroups': []},
+            }
+        ),
+    ),
+    # Test case with custom settings and proxy configuration
+    TestCase(
+        settings=Settings(
+            {
+                'SCHEDULER': 'scrapy.core.scheduler.Scheduler',
+                'ITEM_PIPELINES': {
+                    'scrapy.pipelines.files.FilesPipeline': 1,
+                },
+                'DOWNLOADER_MIDDLEWARES': {
+                    'scrapy.downloadermiddlewares.robotstxt.RobotsTxtMiddleware': 123,
+                    'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 234,
+                    'scrapy.downloadermiddlewares.retry.RetryMiddleware': 345,
+                    'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware': 543,
+                },
+            }
+        ),
+        proxy_config={'useApifyProxy': True, 'apifyProxyGroups': []},
+        expected_output=Settings(
+            {
+                'SCHEDULER': 'apify.scrapy.scheduler.ApifyScheduler',
+                'ITEM_PIPELINES': {
+                    'scrapy.pipelines.files.FilesPipeline': 1,
+                    'apify.scrapy.pipelines.ActorDatasetPushPipeline': 1000,
+                },
+                'DOWNLOADER_MIDDLEWARES': {
+                    'scrapy.downloadermiddlewares.robotstxt.RobotsTxtMiddleware': None,
+                    'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': None,
+                    'scrapy.downloadermiddlewares.retry.RetryMiddleware': None,
+                    'apify.scrapy.middlewares.ApifyHttpProxyMiddleware': 950,
+                    'apify.scrapy.middlewares.ApifyRetryMiddleware': 1000,
+                    'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware': 543,
+                },
+                'APIFY_PROXY_SETTINGS': {'useApifyProxy': True, 'apifyProxyGroups': []},
+            }
+        ),
+    ),
+]
+
+
+@pytest.mark.parametrize('tc', test_cases)
+@pytest.mark.only()
```

Comment:
```suggestion
```
Leftover?

Reply:
yeah... thanks

Reply:
I made https://github.com/fnesveda/flake8-no-pytest-mark-only exactly for this reason but then we migrated to ruff 😅